File: SOLID/DependencyInversion/infrastructure/AvailabilityDaoImpl.cs
namespace SOLID.DependencyInversion.infrastructure
{
    public class AvailabilityDaoImpl : IAvailabilityDao {
    
        public virtual bool IsAvailable() {
            //En realite il y aurait une dependance vers une base de donnéesS...
            return true; 
        }
    
    }
}

File: Java/Exercices/workspaces/solid-dip-start/src/main/java/zenika/solid/dip/infrastructure/AvailabilityDaoImpl.java
package zenika.solid.dip.infrastructure;

public class AvailabilityDaoImpl implements AvailabilityDao {
    
    @Override public boolean isAvailable() {
        //En realite il y aurait une dependance vers JPA/Hib/un WS...
        return true; 
    }
    
}
